src/core/contracts/identity.ts

```diff
@@ -377,45 +380,95 @@ export const removeDelegate = async (
 };
 
 /**
- * getAddDelegateLogs() Retrieves event filter for DSNPAddDelegate event
+ * getDelegeteIdentitiesFor() Retrieves all active identities for an address
  *
- * @param ethereumAddress - Address to filter on
+ * @param ethereumAddress - The address to find all active identities associated to it
  * @param opts - Optional. Configuration overrides, such as from address, if any
- * @returns DSNPBatch event filter
  */
-export const getAddDelegateLogData = async (
+export const getDelegeteIdentitiesFor = async (
   ethereumAddress: EthereumAddress,
   opts?: ConfigOpts
-): Promise<DelegateAddLogData[]> => {
-  const eventSignature = hash("DSNPAddDelegate(address,uint8)");
-  const addDelegateLogs = await getDelegateLogs(ethereumAddress, eventSignature, opts);
-  const addDelegateLogData: DelegateAddLogData[] = siftDelegateLogs(addDelegateLogs);
+): Promise<EthereumAddress[]> => {
+  const delegateLogData: DelegateLogData[] = siftDelegateLogs(await getAllDelegateLogsFor(ethereumAddress, opts));
+
+  const activeIdentities = (await resolveDelegatesFor(ethereumAddress, delegateLogData)).map(
+    (data: DelegateLogData) => data?.identityAddress
+  );
 
-  return addDelegateLogData;
+  return activeIdentities;
 };
 
 /**
- * getRemoveDelegateLogs() Retrieves event filter for DSNPRemoveDelegate event
+ * resolveDelegatesFor() Filter delegates that have not been removed
  *
- * @param ethereumAddress - Address to filter on
+ * @param ethereumAddress - The address to find all active identities associated to it
+ * @param delegateLogData - Log data for delegates (DSNPAddDelegate and DSNPRemoveDelegate)
  * @param opts - Optional. Configuration overrides, such as from address, if any
- * @returns DSNPBatch event filter
  */
-export const getRemoveDelegateLogData = async (
+export const resolveDelegatesFor = async (
   ethereumAddress: EthereumAddress,
+  delegateLogData: DelegateLogData[],
   opts?: ConfigOpts
-): Promise<DelegateRemoveLogData[]> => {
-  const eventSignature = hash("DSNPRemoveDelegate(address,uint64)");
-  const removeDelegateLogs = await getDelegateLogs(ethereumAddress, eventSignature, opts);
-  const removeDelegateLogData = siftDelegateLogs(removeDelegateLogs) as DelegateRemoveLogData[];
+): Promise<DelegateLogData[]> => {
+  const provider = requireGetProvider(opts);
+
+  const currentBlockNumber = await provider.getBlockNumber();
+  const isRemoved = (delegate: DelegateRemoveLogData, currentBlockNumber: number): boolean =>
+    delegate.endBlock <= currentBlockNumber;
+
+  const groupedByIdentityAddress = groupByIdentityAddress(delegateLogData);
 
-  return removeDelegateLogData;
+  const activeDelegates = Object.values(groupedByIdentityAddress).reduce<DelegateLogData[]>(
```

Comment:
nit: this is log data but it's called "activeDelegates" which makes me at first think they are addresses. The return is DelegateLogData type. Please rename these variables to indicate it's log data so people (me, perhaps your future self 😁  ) don't have to do a re-read to clear up initial confusion.

Reply:
nice. I love comments that improve readability.